### ISSUE TITLE
Right-clicking a file should not open, should only show context menu

### ISSUE THREAD

**Issue by internationils** [RickStrahl/MarkdownMonster#1112]
Right-clicking a file in the explorer view should not open, should only show context menu. Sometimes I just want to rename a file or something else from the context menu. The left button does the opening, right should only be context.

**Comment by RickStrahl** [comment]
Agree. Handling selections is really tricky unfortunately with changes potentially having lots of side effects because there are so many combinations of operations related to clicks. 

If I recall the decision was made explicitly originally because it severely complicated things by not moving the selection.

Taking a look around various applications most - but not all - do what you suggest, ie. they don't navigate.

**Comment by RickStrahl** [comment]
Ok I was able to make this work. 

I also added the filename to the top of the context menu so there's no confusion what you're working with.

![image](https://github.com/RickStrahl/MarkdownMonster/assets/1374013/02430d9c-0100-4e54-bb2b-5c74bcc82654)

Fix in 3.2.17.1

**Comment by internationils** [comment]
Mostly fixed... if you right click and get the context menu, and then accidentally right click on the file while the context window is open (it was an honest misclick, I promise!! : ) then it still shows up in the preview.

**Comment by RickStrahl** [comment]
Hmmm... damn this is complex. The issue is that in order to handle all these special click scenarios MM does low level mouse and key handling and there 3 different mouse events that interact.

Played around with this with rapid clicks and 'misclicks' and sure enough I was able to occasionally right click and open a document. Still couldn't quite make out what the issue was.

But after a bit of sleuthing I finally figured out that the MouseUp event was still firing and depending on how quickly the button was released. So it was working **most of the time** but not always.

IAC I think I have the fix for this now:


```cs
// in MouseUp
if (e.ChangedButton == MouseButton.Right || e.RightButton == MouseButtonState.Pressed)
{
    isDoubleClick = false;
    return;
}
```

Incidentally the 'misclicks' would also result in crashes on many occasions, so this should fix that as well. I think this fixes potentially a number of related issues where double clicks and/or right clicks are involved which should improve behavior of the folder view in general.

which captures both scenarios.
